site/docs/plugins/session.md

```diff
@@ -636,6 +554,286 @@ For example, the storage adapter for Supabase can be imported from `https://deno
+Without sessions, this would happen:
+
+1. Update with new text message is sent to your bot
+2. No handler is invoked, so no action is taken
+3. The middleware completes immediately
+
+As soon as you install (default, strict) sessions, which directly provide the session data on the context object, this happens:
+
+1. Update with new text message is sent to your bot
+2. Session data is loaded from session storage (e.g. database)
+3. No handler is invoked, so no action is taken
+4. Identical session data is written back to session storage
+5. The middleware completes, and has performed a read and a write to the data storage
+
+Depending on the nature of your bot, this may lead to a lot of superfluous reads and writes.
+Lazy sessions allow you to skip steps 2. and 4. if it turns out that no invoked handler needs session data.
+In that case, no data will be read from the data storage, nor written back to it.
+
+This is achieved by intercepting access to `ctx.session`.
+If no handler is invoked, then `ctx.session` will never be accessed.
+Lazy sessions use this as an indicator to prevent database communication.
+
+In practice, instead of having the session data available under `ctx.session`, you will now have _a promise of the session data_ available under `ctx.session`.
+
+```ts
+// Default sessions (strict sessions)
+bot.command("settings", (ctx) => {
+  // `session` is the session data
+  const session = ctx.session;
+});
+
+// Lazy sessions
+bot.command("settings", async (ctx) => {
+  // `promise` is a Promise of the session data, and
+  const promise = ctx.session;
+  // `session` is the session data
+  const session = await ctx.session;
+});
+```
+
+If you never access `ctx.session`, no operations will be performed, but as soon as you access the `session` property on the context object, the read operation will be triggered.
+If you never trigger the read (or directly assign a new value to `ctx.session`), we know that we also won't need to write any data back, because there is no way it could have been altered.
+Consequently, we skip the write operation, too.
+As a result, we achieve minimal read and write operations, but you can use session almost identical to before, just with a few `async` and `await` keywords mixed into your code.
+
+So what is necessary to use lazy sessions instead of the default (strict) ones?
+You mainly have to do three things:
+
+1. Flavor your context with `LazySessionFlavor` instead of `SessionFlavor`.
+   They work the same way, just that `ctx.session` is wrapped inside a promise for the lazy variant.
+2. Use `lazySession` instead of `session` to register your session middleware.
+3. Always put an inline `await ctx.session` instead of `ctx.session` everywhere in your middleware, for both reads and writes.
+   Don't worry: you can `await` the promise with your session data as many times as you want, but you will always refer to the same value, so there are never going to be duplicate reads for an update.
```

Comment:
```suggestion
   Don't worry: You can `await` the promise with your session data as many times as you want, but you will always refer to the same value, so they are never going to be duplicate reads for an update.
```

Reply:
There are not going to be duplicate reads. Duplicate reads will not happen.

Reply:
Shoud we revert?